src/ray/core_worker/common.h

```diff
@@ -73,10 +73,24 @@ struct TaskInfo {
   const TaskType task_type;
 };
 
-enum class StoreProviderType { PLASMA };
+enum class StoreProviderType { LOCAL_PLASMA, PLASMA };
 
 enum class TaskTransportType { RAYLET };
 
+struct EnumClassHash {
+  template <typename T>
+  std::size_t operator()(T t) const {
+    return static_cast<std::size_t>(t);
+  }
+};
+
+template <typename Key>
+using HashType = typename std::conditional<std::is_enum<Key>::value, EnumClassHash,
+                                           std::hash<Key>>::type;
+
+template <typename Key, typename T>
+using UnorderedMap = std::unordered_map<Key, T, HashType<Key>>;
```

Comment:
This is nice. should we put it in `util.h`? So something else can also use this.
Also, do you think it's better to name it something like `EnumMap` or `EnumUnorderedMap`? 
And I think the `HashType` isn't very useful. Because for non-enum keys, people would just use `std::unordered_map`.

Reply:
makes sense. thanks.